OpenQA/Isotovideo/Utils.pm

```diff
@@ -53,6 +55,40 @@ sub git_remote_url ($git_repo_dir) {
     return 'UNKNOWN (error on git remote call)';    # uncoverable statement
 }
 
+sub _lock_cache_directory ($cache_dir) {
+    open(my $lock, '>', "$cache_dir.lock") or die "Unable to open lock file '$cache_dir.lock' for Git caching: $!\n";
+    flock($lock, LOCK_EX) or die "Unable to lock '$cache_dir.lock' for Git caching: $!\n";
+    return scope_guard sub {
+        flock($lock, LOCK_UN) or die "Unable to unlock '$cache_dir.lock' after Git caching: $!\n";
```

Comment:
Maybe a minor issue but it feels wrong to repeatedly use $cache_dir.lock instead of one `my $lock_file = "$cache_dir.lock";` that can be re-used
```suggestion
    my $lock_file = "$cache_dir.lock";
    open(my $lock, '>', "$lock_file") or die "Unable to open lock file '$lock_file' for Git caching: $!\n";
    flock($lock, LOCK_EX) or die "Unable to lock '$lock_file' for Git caching: $!\n";
    return scope_guard sub {
        flock($lock, LOCK_UN) or die "Unable to unlock '$lock_file' after Git caching: $!\n";
```